src/rustup.rs

```diff
@@ -217,6 +219,7 @@ pub fn rustup_download_list(
             .flat_map(|(_, pkg)| {
                 pkg.target
                     .into_iter()
+                    .filter(|(name, _)| { platforms.contains(&name) || platforms_exe.contains(&name) } )
```

Comment:
As platform list is moved in configuration files, this list is now a user input (and not compiled in binary).

Thus, with your modification there is no runtime-check that a user-provided platform (through mirror.toml) exist. As a result, panamax will report a success even when user ask to develop that non-existent platform.

Reply:
Hello (I worked with @Hyask on the code in this MR),

Should we maybe keep the list of supported platforms in the code, and check the provided input list against this list of supported platforms reporting an error in the case the input configuration is incorrect?

Reply:
As a simple user, I would a least expect a log message if my user input seems to be inexistent.

From my point of view (note that I'm not-related with panamax maintenance, just a single user), your code introduce a big breaking change in Panamax configuration:
- platform and platform_exe are now mandatory in the configuration files ;
- when updating panamax, admin will have to change configuration file in their repo ==> most lazy admin who don't read changelog carefully will have to deal with an unexpected error message and most probably won't think they have to review their mirror.toml file because of new mandatories fields.

==> Maybe you can make this new rustup section field optional and, if not available, fallback to the "good'old" behaviour the sysadmin is used to...

Reply:
This looks like a good point (albeit a bit different from handling incorrect configuration), thank you for your input!

We will update the MR to:

- Make the setting optional, and keep the existing behavior if the option is absent (or the list empty). Whether or not we'd like to put an informational message in that case, informing that all configurations will be downloaded, depends on what the maintainer(s) think(s) is best.
- When the list is populated by the user through the configuration, check that the input platforms exist and are known of panamax, and emit an error if it not the case.